router/internal/persistedoperation/cache.go

```diff
@@ -0,0 +1,31 @@
+package persistedoperation
+
+import (
+	"github.com/dgraph-io/ristretto"
+	"github.com/wundergraph/cosmo/router/internal/unsafebytes"
+)
+
+const (
+	persistentAverageCacheEntrySize = 4 * 1024 // 4kb
+)
+
+type OperationsCache struct {
+	// cache is the backing store for the in-memory cache. Note
+	// that if the cache is disabled, this will be nil
+	Cache *ristretto.Cache[string, []byte]
+}
+
+func (c *OperationsCache) key(clientName string, operationHash []byte) string {
+	return clientName + unsafebytes.BytesToString(operationHash)
```

Comment:
I think it would be better to use uint64 as key, using xxhash to generate it.
You're concatenating clientName with an unsafe string, which allocates, but it's also dangerous because the operationHash must not be changed.

Reply:
Fixed